From b023af68a63f7ef9725cf39db67bd5d63e8440c4 Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Thu, 5 Sep 2013 10:18:45 +0800
Subject: [PATCH 3/3] ENGR00276102-1 fix Music pop up stop issue

Previous fix has some side effect, audio may play twice,
revert previous fix and apply the new fix.

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 src/com/android/music/MediaPlaybackService.java | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/src/com/android/music/MediaPlaybackService.java b/src/com/android/music/MediaPlaybackService.java
index c5cc962..25d60c7 100644
--- a/src/com/android/music/MediaPlaybackService.java
+++ b/src/com/android/music/MediaPlaybackService.java
@@ -1085,17 +1085,12 @@ public class MediaPlaybackService extends Service {
     }
 
     private void setNextTrack() {
-       int pos = getNextPosition(false);
-        if (pos < 0) {
-            gotoIdleState();
-            if (mIsSupposedToBePlaying) {
-               mIsSupposedToBePlaying = false;
-               notifyChange(PLAYSTATE_CHANGED);
-            }
-        } else {
-            mNextPlayPos = pos;
+        mNextPlayPos = getNextPosition(false);
+        if (mNextPlayPos >= 0) {
             long id = mPlayList[mNextPlayPos];
             mPlayer.setNextDataSource(MediaStore.Audio.Media.EXTERNAL_CONTENT_URI + "/" + id);
+        } else {
+            mPlayer.setNextDataSource(null);
         }
     }
 
-- 
1.8.0

